src/scvi/nn/_base_components.py

```diff
@@ -135,13 +145,18 @@ def _hook_fn_zero_out(grad):
                     b = layer.bias.register_hook(_hook_fn_zero_out)
                     self.hooks.append(b)
 
-    def forward(self, x: torch.Tensor, *cat_list: int):
+    def forward(
+        self, x: torch.Tensor, cat_list: list | None = None, cont: torch.Tensor | None = None
```

Comment:
I changed the forward parametrization from `(x, *cat_list)` to `(x,cat_list,cont)` - so this will break with other `Modules` using `FCLayers` by passing `cat_list` in expanded `*` format. So either the parametrization here needs to be fixed or the use of `FCLayers` elsewhere.

Reply:
See my comment in https://github.com/scverse/scvi-tools/issues/3021

Reply:
Thanks. I thought I would be able to finish this in one shot for all models, but its a bit more complex than I thought. So, I'll be focusing on scvi and scanvi for now only.